app/scripts/controllers/transactions/lib/util.js

```diff
@@ -49,22 +55,51 @@ export function validateTxParams(txParams) {
     );
   }
 
-  validateFrom(txParams);
-  validateRecipient(txParams);
-  if ('value' in txParams) {
-    const value = txParams.value.toString();
-    if (value.includes('-')) {
-      throw ethErrors.rpc.invalidParams(
-        `Invalid transaction value "${txParams.value}": not a positive number.`,
-      );
-    }
+  // console.log(txParams);
+
+  Object.keys(txParams).forEach((key) => {
+    const value = txParams[key];
+    // validate types
+    switch (key) {
+      case 'from':
+        validateFrom(txParams);
+        break;
+      case 'to':
+        validateRecipient(txParams);
+        break;
+      case 'value':
+        if (typeof value !== 'string') {
+          throw ethErrors.rpc.invalidParams(
+            `Invalid transaction params: ${key} is not a string. got: (${value})`,
+          );
+        }
+        if (value.toString().includes('-')) {
+          throw ethErrors.rpc.invalidParams(
+            `Invalid transaction value "${value}": not a positive number.`,
+          );
+        }
 
-    if (value.includes('.')) {
-      throw ethErrors.rpc.invalidParams(
-        `Invalid transaction value of "${txParams.value}": number must be in wei.`,
-      );
+        if (value.toString().includes('.')) {
+          throw ethErrors.rpc.invalidParams(
+            `Invalid transaction value of "${value}": number must be in wei.`,
+          );
+        }
+        break;
+      case 'chainId':
+        if (typeof value !== 'number' && typeof value !== 'string') {
+          throw ethErrors.rpc.invalidParams(
+            `Invalid transaction params: ${key} is not a Number or hex string. got: (${value})`,
+          );
+        }
+        break;
+      default:
+        if (typeof value !== 'string') {
+          throw ethErrors.rpc.invalidParams(
```

Comment:
I see that additional validation was added throughout this function, but it looks to be identical to the validation already happening in `tx-state-manager`. Seems reasonable :+1: